@here/olp-sdk-core/lib/utils/multipartupload-internal/NodeFileData.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { BlobData, fs } from "@here/olp-sdk-core";
+
+/**
+ * @internal
+ * Implementation of reading bytes from file by filepath
+ * in Node.js.
+ */
+export class NodeFileData implements BlobData {
+    private fileHandle?: fs.promises.FileHandle;
+
+    constructor(private readonly filePath: string) {}
+
+    async readBytes(offset: number, count: number): Promise<ArrayBuffer> {
+        if (!this.fileHandle) {
+            this.fileHandle = await this.openFile();
+        }
+        const fileSize = await this.size();
+        const bufferLength =
+            offset + count > fileSize ? fileSize - offset : count;
+
+        const buffer = Buffer.alloc(bufferLength);
+        const chunk = await this.fileHandle.read(
+            buffer,
+            0,
+            bufferLength,
+            offset
+        );
+        return chunk.buffer;
+    }
+
+    async size(): Promise<number> {
+        if (!this.fileHandle) {
+            this.fileHandle = await this.openFile();
+        }
+        const stat = await this.fileHandle.stat();

```

Comment:
I would consider doing this in the constructor and keeping the size as a member. In case the promise does not solve i would keep 0 as size.

Reply:
We don't have possibility to use async operations in constructor. In case of we calling the size method first time (we don't have cached file descriptor), we should open the file in async way